src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -406,14 +406,14 @@ internal sealed override ParameterSymbol? ExtensionParameter
                 {
                     var methodSymbol = getMarkerMethodSymbol(@this, uncommon);
 
-                    // Tracked by https://github.com/dotnet/roslyn/issues/76130 : do we want to tighten the flags check further? (require that type be sealed?)
-                    if (methodSymbol.DeclaredAccessibility != Accessibility.Private ||
+                    if (!methodSymbol.ContainingType.IsSealed ||
```

Comment:
>!methodSymbol.ContainingType.IsSealed || [](http://example.com/codeflow?start=24&length=40)

It feels like the check is misplaced. If we say that extension block must be sealed, then we shouldn't treat the enclosing type as ```IsExtension```, and, therefore, should never get here.

Reply:
I wasn't sure because we've kept the checks that make the nested type not count as an extension to the minimum (typekind check and existence of a method with right name). Most detailed checks are delayed and don't mean `IsExtension` is false. I'm good either way